### PR TITLE
Add appointment command

### LINKED ISSUE #93: As a home-based beauty salon owner, I want to add appointments with my clients
I want to add appointments with my clients so that I can keep track of when my customers are scheduled for my services.

### LINKED ISSUE #106: As a Home-based Beauty Salon Owner, I want to be able to list all my appointments so that I can see which are my upcoming ones.

### LINKED ISSUE #107: As a Home-based Beauty Salon Owner, I want to be able to find a particular appointment so that I see the specific details of that appointment.

### LINKED ISSUE #108: As a Home-based Beauty Salon Owner, I want to be able to delete a particular appointment to account for customers who cancel.

### LINKED ISSUE #109: As a Home-based Beauty Salon Owner, I want to be able to clear all appointments if I want to reset Homerce.

### LINKED ISSUE #127: As a Home-based Beauty Salon Owner, I want to be able to mark an appointment as done.

### LINKED ISSUE #128: As a Home-based Beauty Salon Owner, I want to be able to mark an appointment as not done in case I marked it as done by accident..

### LINKED ISSUE #129: As a Home-based Beauty Salon Owner, I want to be able to edit an appointment's details.

### PR DESCRIPTION
**Pull request by khoongwk**
Major update to include classes of Storage, Logic, Ui, and Model for the Appointment class.

Current status: `addappt` command works in the GUI. Entries are stored into `appointments.json`.

However, there is an issue which was uncovered: When a duplicate appointment is added into 
Homerce, the existing `appointments.json` is cleared.

### CODE REVIEW

File: src/main/java/seedu/address/model/manager/ServiceManager.java [hunk 1]
```diff
@@ -8,25 +8,26 @@
...
  */
 public class ServiceManager implements ReadOnlyServiceManager {
 
-    private final UniqueList<Service> services;
+    private final UniqueServiceList services;
 
     public ServiceManager() {
-        this.services = new UniqueList<>();
+        this.services = new UniqueServiceList();
     }
```
Comment (galvinleow) [inline src/main/java/seedu/address/model/manager/ServiceManager.java:24]: I think Hans made a generic list?

Comment (khoongwk) [reply src/main/java/seedu/address/model/manager/ServiceManager.java:24]: Yup he did, but I needed to extend it with more methods. I need a contains method with parameter ServiceCode to check if the service code belongs to an existing service. I can't just insert that as a generic method to Unique List, so I created UniqueServiceList to extend UniqueList<Service>.

**Review by hanstirtaputra — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/logic/parser/appointment/FindAppointmentCommandParser.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            PREFIX_NAME)) {
+            throw new ParseException(MULTIPLE_PARAMETERS);
+        }
+        if ((!arePrefixesPresent(argMultimap, PREFIX_DATE)
+            && !arePrefixesPresent(argMultimap, PREFIX_PHONE)
+            && !arePrefixesPresent(argMultimap, PREFIX_NAME)
+            && !arePrefixesPresent(argMultimap, PREFIX_SERVICE_SERVICE_CODE))
+            || !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                AddAppointmentCommand.MESSAGE_USAGE));
```
Comment (galvinleow) [inline src/main/java/seedu/address/logic/parser/appointment/FindAppointmentCommandParser.java]: FindAppointmentCommand

Comment (khoongwk) [reply src/main/java/seedu/address/logic/parser/appointment/FindAppointmentCommandParser.java]: Fixed, thanks!

**Review by RuiFengg — APPROVED** [review]
LGTM

**Pull request merged**